Moved handle Click logic to impl, added link support in topic detail screen

varunasingh opened the pull request.

varunasingh committed:
Moved handle Click logic to impl, added link support in topic detail
screen.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/DiscussionPostDescriptionRecyclerAdapter.kt:
```diff
@@ -1,45 +1,98 @@
+
+    }
+
+    val onClickListener: BetterLinkMovementMethod.OnLinkClickListener =
```
Why not have a class that does this? e.g.

```
class BetterLinkMovementLinkClickListener(val systemImpl: UstadMobileSystemImpl)
```

mikedawson replied:
e.g. class could then be used by all components that need to listen (instead of repeating the same anonymous listeners)?

mikedawson commented on core/src/androidMain/kotlin/com/ustadmobile/core/impl/UstadMobileSystemImpl.kt:
```diff
@@ -485,6 +485,20 @@ actual open class UstadMobileSystemImpl : UstadMobileSystemCommon() {
         (context as Context).startActivity(intent)
     }
 
+    actual fun handleClickLink(url: String, context: Any){
```
This code should be in common. The logic will be the same.

Needs jvm tests.

Needs to handle edge case: when link is an ustad link, but a different endpoint

varunasingh committed:
Moved movement to own class, moved handleClick To
UstadMobileSystemCommon, Added support for deeplink if link on another
endpoint. Tests wip

varunasingh committed:
Fix taiga issue 760 where students were able to see other people when
searrching people in chat list

varunasingh committed:
Fix Taiga 761 issue.

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/BetterLinkMovementClickListener.kt:
```diff
@@ -0,0 +1,48 @@
+        }
+
+
+    fun addMovement(textView: TextView?){
```
Why should the parameter be nullable?

mikedawson commented on app-android/src/main/java/com/ustadmobile/port/android/view/MessagesRecyclerAdapter.kt:
```diff
@@ -47,7 +52,9 @@ class MessagesRecyclerAdapter(
-        addMovement(holder.binding.itemCommentsListLine2Text)
+        val listener = BetterLinkMovementLinkClickListener(systemImpl, accountManager, context)
+        listener.addMovement(holder.binding.itemCommentsListLine2Text)
+        //addMovement(holder.binding.itemCommentsListLine2Text)
```
Commented out line

varunasingh committed:
1. Remove New Chat Group (not implemented)
2. Upated chat search to search existing chats with people as well as
people not chatting with.
3. Fixed new chat issue in person detail

varunasingh committed:
Removed commented out

varunasingh committed:
rEmoved old test, updated link to use ustadurlcomponent

varunasingh committed:
Fix Taiga issue 773 internaly and on the view fragment

varunasingh committed:
Fix

mikedawson committed:
Merge branch 'primary' into dev-discussion-checklist-fix

mikedawson committed:
Update new RecyclerAdapters to attempt to avoid memory leak. Update SystemImplCommon to remove a few redundant lines.

mikedawson committed:
Fix compilation issues on new recycler adapter (nullable context var)
